go/vt/sqlparser/sql.y

```diff
@@ -1144,6 +1151,11 @@ table_column_list:
   {
     $$.AddColumn($3)
   }
+| table_column_list ',' column_definition_for_create constraint_definition
```

Comment:
This is only valid for check constraints, right? Not for foreign keys

Reply:
Add a test

Reply:
Probably need to split check_constraint_definition into its own rule for this

Reply:
fix and invalidSQL test added